### DISCUSSION TITLE
I cant open jbig2 file in  Magick.NET

### DISCUSSION THREAD

**Discussion by Charltsing** [dlemstra/Magick.NET#1380] [Help] [answered]
I dont find JPEG2 format in MagickFormat.cs
if i open jbig2 file ,it report:

    ImageMagick.MagickMissingDelegateErrorException
        HResult=0x80131500
        Message=no decode delegate for this image format `JBG' @ error/constitute.c/ReadImage/737
        Source=Magick.NET-Q16-AnyCPU

*****************************************

but ImageMagick support JBEG2
https://hub.njuu.cf/ImageMagick/ImageMagick/issues/623#issuecomment-487014867
On the official 7.0.8-42 (Windows Q16-x64) build:
> magick -list compress
...
JBIG1
JBIG2
JPEG2000
JPEG

*****************************************
maybe some code can help you :
[JBig2Decoder.NET](https://github.com/Ragnarokkr-Xia/CondenseTech.JBig2Decoder) ported to .Net 5.0/6.0/7.0 and .Net Standard 2.0/2.1
[JBig2Decoder.NET](https://github.com/winkingsoftware/JBig2Decoder.NETCore)  A .NET Core version by Winking of the JBig2Decoder.NET package by devteamexpress

****************************************
you can found jbig and jbig2 sample image :
https://www.cl.cam.ac.uk/~mgk25/jbigkit/download/jbigkit-2.1.tar.gz
https://jbig2dec.com/tests/index.html

****************************************

**Comment by dlemstra** [comment]
That library is not supported because this is licensed under GPL 2.0 and that is not compatible with the license of this project. ImageMagick does know that the compression exists but that doesn't mean it can read it.

**Reply by Charltsing** [reply]
Maybe we should write our own encoding and decoding library, use C# ?

**Comment by Charltsing** [comment]
https://hub.njuu.cf/afila/JBIG2-Image-Decoder.NET

 this is a winform sample

**Comment by dlemstra** [comment]
I will not be adding support for this to the library because this would mean changing ImageMagick and we would need to find another library or implement it ourselves and that will probably not happen.

My advise would be to use another library to read the jbig2 image and then convert it to something that can be read by Magick.NET. It looks like JBig2Decoder.NETCore can decode the buffer into another format and then you can feed the byte array to Magick.NET.

**Comment by Charltsing** [comment, marked as answer]
[JBIG2Viewer.zip](https://github.com/dlemstra/Magick.NET/files/11394143/JBIG2Viewer.zip)
